index.js

```diff
@@ -255,6 +255,71 @@ exports.Transform = class Transform extends stream.Transform {
 
 exports.PassThrough = class PassThrough extends exports.Transform {}
 
+exports.finished = function finished (stream, opts, cb) {
+  const defaultOpts = {
+    error: true,
+    readable: true,
+    writable: true,
+    cleanup: false
+  }
+
+  if (typeof opts === 'object' && opts !== null) {
+    opts = { ...defaultOpts, ...opts }
+  }
+
+  if (typeof opts === 'function') {
+    cb = opts
+    opts = defaultOpts
+  }
```

Comment:
```suggestion
  if (typeof opts === 'function') {
    cb = opts
    opts = {}
  }

  if (!opts) opts = {}
  
  const {
    error = true,
    readable = stream.readable,
    writable = stream.writable,
    cleanup = false
  } = opts
```

This is a general pattern in JavaScript when dealing with overloaded functions:

1. First, shuffle `cb` into place, zeroing the overloaded arguments along the way.
2. Then, zero any possibly nullish arguments.
3. Finally, destructure the `opts` argument with appropriate defaults.

This change will cause issues with the following code, I'll tweak bits as necessary.

Reply:
thanks for documenting here the steps